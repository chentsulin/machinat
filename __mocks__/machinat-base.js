import moxy from 'moxy';

const baseModule = jest.requireActual('machinat-base');

module.exports = {
  ...baseModule,
  BaseBot: moxy(baseModule.BaseBot, { mockProperty: false }),
};
