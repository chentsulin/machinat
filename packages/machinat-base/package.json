{
  "name": "machinat-base",
  "version": "0.0.7",
  "author": "Patrick Liu <zeldalink0515@gmail.com>  (https://github.com/lrills)",
  "repository": {
    "type": "git",
    "url": "https://github.com/lrills/machinat.git",
    "directory": "packages/machinat-base"
  },
  "homepage": "https://github.com/lrills/machinat#readme",
  "bugs": "https://github.com/lrills/machinat",
  "main": "lib/index.js",
  "files": [
    "lib/*",
    "README.md"
  ],
  "license": "MIT",
  "dependencies": {
    "delay": "^4.1.0",
    "invariant": "^2.2.4",
    "machinat-utility": "^0.0.7",
    "raw-body": "^2.3.3",
    "symbol-observable": "^1.2.0",
    "thenifiedly": "^0.1.0",
    "warning": "^4.0.2"
  },
  "peerDependencies": {
    "machinat": "^0.0.1"
  }
}
