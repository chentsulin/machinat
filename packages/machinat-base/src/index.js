// @flow
export { default as BaseConnector } from './baseConnector';
export { default as HTTPConnector } from './httpConnector';
export { default as HttpContext } from './httpContext';
export { default as Bot } from './bot';
export * from './eventUtils';
