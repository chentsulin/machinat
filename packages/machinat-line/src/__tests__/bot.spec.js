import { WebhookConnector } from 'machinat-base';
import LineBot from '../bot';
import Client from '../client';

it('throws if accessToken not given', () => {
  expect(() => new LineBot()).toThrowErrorMatchingInlineSnapshot(
    `"should provide accessToken to send messenge"`
  );
});

it('throws if shouldValidateRequest but channelSecret not given', () => {
  expect(
    () => new LineBot({ accessToken: '_TOKEN_', shouldValidateRequest: true })
  ).toThrowErrorMatchingInlineSnapshot(
    `"should provide channelSecret if shouldValidateRequest set to true"`
  );
});

it('is ok to have channelSecret empty if shouldValidateRequest set to false', () => {
  expect(
    () => new LineBot({ accessToken: '_TOKEN_', shouldValidateRequest: false })
  ).not.toThrow();
});

it('creates client and connector', () => {
  const bot = new LineBot({
    accessToken: '_TOKEN_',
    channelSecret: '_SECRET_',
  });

  expect(bot.client).toBeInstanceOf(Client);
  expect(bot.connector).toBeInstanceOf(WebhookConnector);
});

it('sets default options', () => {
  expect(
    new LineBot({ accessToken: '_TOKEN_', channelSecret: '_SECRET_' }).options
  ).toMatchInlineSnapshot(`
Object {
  "accessToken": "_TOKEN_",
  "channelSecret": "_SECRET_",
  "connectionCapicity": 100,
  "shouldValidateRequest": true,
  "useReplyAPI": false,
}
`);
});

it('covers default options', () => {
  const options = {
    accessToken: '_TOKEN_',
    shouldValidateRequest: false,
    channelSecret: '_SECRET_',
    connectionCapicity: 9999,
    useReplyAPI: true,
  };
  expect(new LineBot(options).options).toEqual(options);
});
