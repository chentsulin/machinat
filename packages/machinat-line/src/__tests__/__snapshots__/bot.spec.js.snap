// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render(token, node, options) make api calls 1`] = `
Object {
  "jobs": Array [
    Object {
      "body": Object {
        "messages": Array [
          Object {
            "text": "Hello LINE",
            "type": "text",
          },
          Object {
            "originalContentUrl": "https://machinat.com/greeting.png",
            "previewImageUrl": undefined,
            "quickReply": Object {
              "items": Array [
                Object {
                  "action": Object {
                    "label": "HI",
                    "text": "Hi!",
                    "type": "message",
                  },
                  "imageUrl": undefined,
                  "type": "action",
                },
              ],
            },
            "type": "image",
          },
        ],
        "to": "john_doe",
      },
      "channelUid": "line:_MY_BOT_:user:john_doe",
      "method": "POST",
      "path": "v2/bot/message/push",
    },
    Object {
      "body": null,
      "channelUid": "line:_MY_BOT_:user:john_doe",
      "method": "POST",
      "path": "v2/bot/user/john_doe/richmenu/newbie",
    },
  ],
  "results": Array [
    Object {},
    Object {},
  ],
  "tasks": Array [
    Object {
      "payload": Array [
        Object {
          "body": Object {
            "messages": Array [
              Object {
                "text": "Hello LINE",
                "type": "text",
              },
              Object {
                "originalContentUrl": "https://machinat.com/greeting.png",
                "previewImageUrl": undefined,
                "quickReply": Object {
                  "items": Array [
                    Object {
                      "action": Object {
                        "label": "HI",
                        "text": "Hi!",
                        "type": "message",
                      },
                      "imageUrl": undefined,
                      "type": "action",
                    },
                  ],
                },
                "type": "image",
              },
            ],
            "to": "john_doe",
          },
          "channelUid": "line:_MY_BOT_:user:john_doe",
          "method": "POST",
          "path": "v2/bot/message/push",
        },
        Object {
          "body": null,
          "channelUid": "line:_MY_BOT_:user:john_doe",
          "method": "POST",
          "path": "v2/bot/user/john_doe/richmenu/newbie",
        },
      ],
      "type": "dispatch",
    },
  ],
}
`;

exports[`#render(token, node, options) works with replyToken 1`] = `
Object {
  "jobs": Array [
    Object {
      "body": Object {
        "messages": Array [
          Object {
            "text": "Hello LINE",
            "type": "text",
          },
          Object {
            "originalContentUrl": "https://machinat.com/greeting.png",
            "previewImageUrl": undefined,
            "quickReply": Object {
              "items": Array [
                Object {
                  "action": Object {
                    "label": "HI",
                    "text": "Hi!",
                    "type": "message",
                  },
                  "imageUrl": undefined,
                  "type": "action",
                },
              ],
            },
            "type": "image",
          },
        ],
        "replyToken": "__REPLY_TOKEN__",
      },
      "channelUid": "line:_MY_BOT_:user:john_doe",
      "method": "POST",
      "path": "v2/bot/message/reply",
    },
    Object {
      "body": null,
      "channelUid": "line:_MY_BOT_:user:john_doe",
      "method": "POST",
      "path": "v2/bot/user/john_doe/richmenu/newbie",
    },
  ],
  "results": Array [
    Object {},
    Object {},
  ],
  "tasks": Array [
    Object {
      "payload": Array [
        Object {
          "body": Object {
            "messages": Array [
              Object {
                "text": "Hello LINE",
                "type": "text",
              },
              Object {
                "originalContentUrl": "https://machinat.com/greeting.png",
                "previewImageUrl": undefined,
                "quickReply": Object {
                  "items": Array [
                    Object {
                      "action": Object {
                        "label": "HI",
                        "text": "Hi!",
                        "type": "message",
                      },
                      "imageUrl": undefined,
                      "type": "action",
                    },
                  ],
                },
                "type": "image",
              },
            ],
            "replyToken": "__REPLY_TOKEN__",
          },
          "channelUid": "line:_MY_BOT_:user:john_doe",
          "method": "POST",
          "path": "v2/bot/message/reply",
        },
        Object {
          "body": null,
          "channelUid": "line:_MY_BOT_:user:john_doe",
          "method": "POST",
          "path": "v2/bot/user/john_doe/richmenu/newbie",
        },
      ],
      "type": "dispatch",
    },
  ],
}
`;

exports[`#renderMulticast(targets, node) make api call to message/mulitcast 1`] = `
Object {
  "jobs": Array [
    Object {
      "body": Object {
        "messages": Array [
          Object {
            "text": "Hello LINE",
            "type": "text",
          },
          Object {
            "originalContentUrl": "https://machinat.com/greeting.png",
            "previewImageUrl": undefined,
            "quickReply": Object {
              "items": Array [
                Object {
                  "action": Object {
                    "label": "HI",
                    "text": "Hi!",
                    "type": "message",
                  },
                  "imageUrl": undefined,
                  "type": "action",
                },
              ],
            },
            "type": "image",
          },
        ],
        "to": Array [
          "john",
          "wick",
          "dog",
        ],
      },
      "method": "POST",
      "path": "v2/bot/message/multicast",
    },
    Object {
      "body": Object {
        "richMenuId": "newbie",
        "userIds": Array [
          "john",
          "wick",
          "dog",
        ],
      },
      "channelUid": undefined,
      "method": "POST",
      "path": "v2/bot/richmenu/bulk/link",
    },
  ],
  "results": Array [
    Object {},
    Object {},
  ],
  "tasks": Array [
    Object {
      "payload": Array [
        Object {
          "body": Object {
            "messages": Array [
              Object {
                "text": "Hello LINE",
                "type": "text",
              },
              Object {
                "originalContentUrl": "https://machinat.com/greeting.png",
                "previewImageUrl": undefined,
                "quickReply": Object {
                  "items": Array [
                    Object {
                      "action": Object {
                        "label": "HI",
                        "text": "Hi!",
                        "type": "message",
                      },
                      "imageUrl": undefined,
                      "type": "action",
                    },
                  ],
                },
                "type": "image",
              },
            ],
            "to": Array [
              "john",
              "wick",
              "dog",
            ],
          },
          "method": "POST",
          "path": "v2/bot/message/multicast",
        },
        Object {
          "body": Object {
            "richMenuId": "newbie",
            "userIds": Array [
              "john",
              "wick",
              "dog",
            ],
          },
          "channelUid": undefined,
          "method": "POST",
          "path": "v2/bot/richmenu/bulk/link",
        },
      ],
      "type": "dispatch",
    },
  ],
}
`;
